lib/api/filter.rb

```diff
@@ -43,13 +42,22 @@ def parse
         end
 
         associations.map! { |assoc| ".#{assoc}" }
-        field = "#{model.name}#{associations.join}-#{attr}"
-        target = parsed_filter[:logical_or] ? or_expressions : and_expressions
-        target << {parsed_filter[:operator] => {"field" => field, "value" => parsed_filter[:value]}}
+
+        field    = "#{model.name}#{associations.join}-#{attr}"
+        operator = parsed_filter[:operator]
+        expr     = if parsed_filter[:value].kind_of?(Array)
+                     {"OR" => parsed_filter[:value].map { |v| single_expression(field, operator, v) }}
+                   else
+                     single_expression(field, operator, parsed_filter[:value])
+                   end
+
+        if parsed_filter[:logical_or]
```

Comment:
this is so much nicer than the previous code.

Reply:
Agreed, I pulled a decent about of the commits from https://github.com/ManageIQ/manageiq-api/pull/1059 because I thought this needed to be done regardless if that purpose of that PR got in.